app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/KeepAccessToSAController.scala

```diff
@@ -72,6 +74,7 @@ class KeepAccessToSAController @Inject()(
                 case Right(true) =>
                   Redirect(routes.SignInWithSAAccountController.view)
                 case Right(false) =>
+                  auditHandler.audit(AuditEvent.auditSuccessfullyEnrolledPersonalTax(false))
```

Comment:
No additional logic has been added here. We always have to do it twice because the redirect location is different depending on the option chose. I also think this audit event belongs to the controller.